pom.xml

```diff
@@ -598,6 +598,13 @@ flexible messaging model and an intuitive client API.</description>
       <artifactId>powermock-module-testng</artifactId>
       <scope>test</scope>
     </dependency>
+
+    <dependency>
+      <groupId>org.projectlombok</groupId>
+      <artifactId>lombok</artifactId>
+      <version>1.16.20</version>
+      <scope>provided</scope>
```

Comment:
do we need to include it in `pulsar-client` module? if scope is `provided` then does it mean it will not be part of binary distribution generation?

Reply:
I think once code is auto-generated, then we may not need it in our build..?

Reply:
https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#Dependency_Scope

Reply:
Correct, this is marked as `provided` since it's only required at build time. No dependency is needed at runtime since the boilerplate code gets automatically generated and that's what gets included in the jar.

Reply:
> I think once code is auto-generated, then we may not need it in our build..?

@rdhabalia  To clarify, the code is generated on the fly at compile time. So the dependency it's needed to build from sources, but it's not needed in the binary dist (or as a runtime dependency).